Usemux

miretskiy opened the pull request:
Draft;

Commit:
changefeedccl: Enable changefeed to use `MuxRangeFeed`

Enable changefeed to use `MuxRangeFeed` RPC if
`changefeed.mux_rangefeed.enabled` is set to true.

Release note (enterprise change): Changefeeds may opt in
via `changefeed.mux_rangefeed.enabled` setting to use
`MuxRangeFeed` RPC which multiplexes multipe range feed streams
onto a single RPC stream per node.

Release justification: add option to enable functionality
that's disabled by default.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/85579)
<!-- Reviewable:end -->

Commit:
kv: Add WithMuxRangeFeed option to rangefeed library

Add `WithMuxRangeFeed` option to rangefeed library to enable
the use of `MuxRangeFeed` RPC when executing rangefeeds.

In addition arrange for this option, along with few others, to
be initialized metamorphically for builds to increase option usage
coverage in tests.

Release justification: non production, test only change.
Release note: None
